File: winter-compiler/src/test/resources/InjectConstructorAnnotation_WinterFactory.java
package test;

import io.jentz.winter.ClassTypeKey;
import io.jentz.winter.Component;
import io.jentz.winter.Graph;
import io.jentz.winter.TypeKey;
import io.jentz.winter.inject.Factory;
import io.jentz.winter.inject.InterOp;
import java.lang.Override;
import java.lang.String;
import javax.annotation.Generated;

@Generated(
        value = "io.jentz.winter.compiler.WinterProcessor",
        date = "2019-02-10T14:52Z"
)
public final class InjectConstructorAnnotation_WinterFactory implements Factory<InjectConstructorAnnotation> {
    @Override
    public InjectConstructorAnnotation invoke(final Graph graph) {
        return new InjectConstructorAnnotation(graph.instanceOrNullByKey(new ClassTypeKey<String>(String.class, null)));
    }

    @Override
    public TypeKey<InjectConstructorAnnotation> register(final Component.Builder builder,
                                                         final boolean override) {
        TypeKey<test.InjectConstructorAnnotation> key = new ClassTypeKey<InjectConstructorAnnotation>(InjectConstructorAnnotation.class, null);
        InterOp.prototype(builder, key, override, this);
        return key;
    }
}

File: winter-compiler/src/test/resources/InjectConstructorAnnotation_WinterFactory.kt
package test

import io.jentz.winter.Component
import io.jentz.winter.Graph
import io.jentz.winter.TypeKey
import io.jentz.winter.inject.Factory
import javax.annotation.Generated
import kotlin.Boolean
import kotlin.String

@Generated(
  value = ["io.jentz.winter.compiler.WinterProcessor"],
  date = "2019-02-10T14:52Z"
)
class InjectConstructorAnnotation_WinterFactory : Factory<InjectConstructorAnnotation> {
  override fun invoke(graph: Graph): InjectConstructorAnnotation =
      InjectConstructorAnnotation(graph.instanceOrNull<String>())

  override fun register(builder: Component.Builder, override: Boolean):
      TypeKey<InjectConstructorAnnotation> = builder.prototype(override = override, factory = this)
}
